### ISSUE TITLE
StreamToLogger does not retain attributes of the underlying stream

### ISSUE THREAD

**Issue by Xarthisius** [girder/girder#2520]
I got a following traceback from a custom plugin:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/cheroot/server.py", line 1152, in communicate
    req.respond()
  File "/usr/local/lib/python3.5/dist-packages/cheroot/server.py", line 974, in respond
    self.server.gateway(self).respond()
  File "/usr/local/lib/python3.5/dist-packages/cheroot/wsgi.py", line 133, in respond
    for chunk in filter(None, response):
  File "/usr/local/lib/python3.5/dist-packages/wsgidav/wsgidav_app.py", line 435, in __call__
    for v in app_iter:
  File "/usr/local/lib/python3.5/dist-packages/wsgidav/debug_filter.py", line 167, in __call__
    sub_app_start_response.exc_info)
  File "/usr/local/lib/python3.5/dist-packages/wsgidav/wsgidav_app.py", line 424, in _start_response_wrapper
    safeReEncode(environ.get("PATH_INFO", ""), sys.stdout.encoding),
AttributeError: 'StreamToLogger' object has no attribute 'encoding'
```

After a bit of digging in the code it turns out that `StreamToLogger` retains only callable attributes of the underlying stream: https://github.com/girder/girder/blob/master/girder/__init__.py#L106

Is there any reason why properties are not used?

**Comment by manthey** [comment]
I can't remember my complete rationale at this time, but it had to do with interacting with wsi.  I know that copying all of the properties caused a problem, so I only copied the methods.

I'm hesitant to just copy all non-private properties.  Maybe we should copy all non-private properties are in a list of specific instance types (strings, booleans, and numbers, for instance).

**Comment by manthey** [comment]
@Xarthisius PR #2522 should fix this for you.

**Comment by manthey** [comment]
Fixed in #2522.

**Comment by HDVUCAIR** [comment]
Has this fix made it into the docker hub image girder/girder:latest (as of 16 Aug)?

I'm seeing errors when invoking the plugin recompile/restart from the web browser admin panel after modifying plugin settings:

  File "/girder/girder/utility/install.py", line 172, in runWebBuild
    quiet = '--no-progress=false' if sys.stdout.isatty() else '--no-progress=true'
AttributeError: 'UnicodeStdout' object has no attribute 'isatty'

I confirmed that the native python sys within the docker container does have an isatty in sys.stdout.  However, it seems that the redefinition of sys.stdout in girder/__init__.py is not capturing this particular attribute.
